[2024-04-23T15:27:03.662+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-23T15:27:03.690+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: model_pipeline.evaluate_model manual__2024-04-23T15:23:56.381355+00:00 [queued]>
[2024-04-23T15:27:03.702+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: model_pipeline.evaluate_model manual__2024-04-23T15:23:56.381355+00:00 [queued]>
[2024-04-23T15:27:03.703+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-23T15:27:03.720+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): evaluate_model> on 2024-04-23 15:23:56.381355+00:00
[2024-04-23T15:27:03.732+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=1490) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-04-23T15:27:03.734+0000] {standard_task_runner.py:63} INFO - Started process 1499 to run task
[2024-04-23T15:27:03.734+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'model_pipeline', 'evaluate_model', 'manual__2024-04-23T15:23:56.381355+00:00', '--job-id', '123', '--raw', '--subdir', 'DAGS_FOLDER/ml_pipeline_dag.py', '--cfg-path', '/tmp/tmpibv0633f']
[2024-04-23T15:27:03.736+0000] {standard_task_runner.py:91} INFO - Job 123: Subtask evaluate_model
[2024-04-23T15:27:03.798+0000] {task_command.py:426} INFO - Running <TaskInstance: model_pipeline.evaluate_model manual__2024-04-23T15:23:56.381355+00:00 [running]> on host ec31c9b90297
[2024-04-23T15:27:03.920+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='model_pipeline' AIRFLOW_CTX_TASK_ID='evaluate_model' AIRFLOW_CTX_EXECUTION_DATE='2024-04-23T15:23:56.381355+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-04-23T15:23:56.381355+00:00'
[2024-04-23T15:27:03.923+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-23T15:27:05.533+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/base.py:493: UserWarning: X does not have valid feature names, but RandomForestClassifier was fitted with feature names
  warnings.warn(

[2024-04-23T15:27:07.743+0000] {xcom.py:677} ERROR - Object of type ndarray is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2024-04-23T15:27:07.746+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-23T15:27:07.747+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/serialization/serde.py", line 189, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'numpy.ndarray'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/ml_pipeline_dag.py", line 103, in evaluate_model
    context['task_instance'].xcom_push(key='y_pred', value=y_pred)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 79, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 247, in set
    value = cls.serialize_value(
            ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/xcom.py", line 675, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type ndarray is not JSON serializable
[2024-04-23T15:27:07.762+0000] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=model_pipeline, task_id=evaluate_model, execution_date=20240423T152356, start_date=20240423T152703, end_date=20240423T152707
[2024-04-23T15:27:07.779+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 123 for task evaluate_model (Object of type ndarray is not JSON serializable; 1499)
[2024-04-23T15:27:07.862+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-04-23T15:27:07.893+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-23T15:27:07.896+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
